internal/tss/channels/webhook_channel.go

```diff
@@ -0,0 +1,77 @@
+package channels
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/alitto/pond"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/wallet-backend/internal/tss"
+	tssutils "github.com/stellar/wallet-backend/internal/tss/utils"
+	"github.com/stellar/wallet-backend/internal/utils"
+)
+
+type WebhookChannelConfigs struct {
+	HTTPClient           utils.HTTPClient
+	MaxBufferSize        int
+	MaxWorkers           int
+	MaxRetries           int
+	MinWaitBtwnRetriesMS int
+}
+
+type webhookPool struct {
+	Pool                 *pond.WorkerPool
+	HTTPClient           utils.HTTPClient
+	MaxRetries           int
+	MinWaitBtwnRetriesMS int
+}
+
+var WebhookChannelName = "WebhookChannel"
+
+var _ tss.Channel = (*webhookPool)(nil)
+
+func NewWebhookChannel(cfg WebhookChannelConfigs) *webhookPool {
+	pool := pond.New(cfg.MaxBufferSize, cfg.MaxWorkers, pond.Strategy(pond.Balanced()))
+	return &webhookPool{
+		Pool:                 pool,
+		HTTPClient:           cfg.HTTPClient,
+		MaxRetries:           cfg.MaxRetries,
+		MinWaitBtwnRetriesMS: cfg.MinWaitBtwnRetriesMS,
+	}
+
+}
+
+func (p *webhookPool) Send(payload tss.Payload) {
+	p.Pool.Submit(func() {
+		p.Receive(payload)
+	})
+}
+
+func (p *webhookPool) Receive(payload tss.Payload) {
+	resp := tssutils.PayloadTOTSSResponse(payload)
+	jsonData, err := json.Marshal(resp)
+	if err != nil {
+		log.Errorf("%s: error marshaling payload: %e", WebhookChannelName, err)
+		return
+	}
+	var i int
+	for i = 0; i < p.MaxRetries; i++ {
+		resp, err := p.HTTPClient.Post(payload.WebhookURL, "application/json", bytes.NewBuffer(jsonData))
```

Comment:
Here we are just sending the data as a JSON body payload to the webhook URL, right? There's no authentication nor any custom header.
I'm imagining here that maybe we should support some kind of `Signature` header, to add a security layer in the communication between the `wallet-backend` and the customer application.
What do you think?

Reply:
I actually do like the idea of adding a security layer, but want to give it some more thought. I'm working on a doc for tss v2 ideas and I'll make sure to add this to it. I think that we should be ok not having auth for now because customer applications will deploy their own wallet backends so there's some inherent trust there already